packages/contracts-bedrock/src/universal/WETH98.sol

```diff
@@ -49,16 +49,26 @@ contract WETH98 is IWETH {
         return "WETH";
     }
 
+    /// @inheritdoc IWETH
+    function allowance(address src, address guy) public view virtual returns (uint256) {
+        return _allowance[src][guy];
+    }
+
+    /// @inheritdoc IWETH
+    function balanceOf(address src) public view returns (uint256) {
+        return _balanceOf[src];
+    }
+
     /// @inheritdoc IWETH
     function deposit() public payable virtual {
-        balanceOf[msg.sender] += msg.value;
+        _balanceOf[msg.sender] += msg.value;
         emit Deposit(msg.sender, msg.value);
     }
 
     /// @inheritdoc IWETH
     function withdraw(uint256 wad) public virtual {
-        require(balanceOf[msg.sender] >= wad);
-        balanceOf[msg.sender] -= wad;
+        require(balanceOf(msg.sender) >= wad);
```

Comment:
i would not make a function call in these cases, i would rather just use the balance to avoid the extra jump. If someone wants to edit the balance of a specific user, that can be accomplished in the constructor of the inheriting contract - the key difference here is that `balanceOf` is the balance of a single user for the entire contract, while `allowance` is the allowance of a user for a specific owner, so to make a global allowance is not as straight forward as updating something in the constructor as it is with `balanceOf`

Reply:
perfect, will use the internal then 👍